documentation/src/main/asciidoc/topics/contributing/terminology.adoc

```diff
@@ -0,0 +1,86 @@
+
+Cache template::
+An artifact from which you create cache configurations. +
+{brandname} provides some pre-defined templates that can be referenced using the `org.infinispan.client.hotrod.DefaultTemplate` enum. +
+Users can define new templates in `infinispan.xml` with the `*-cache-configuration` tags. +
+In some APIs, you can use the configuration of an existing cache as a template to configure a new cache. +
+Users can also define templates in `infinispan.xml` with the `*-cache-configuration` tags.
+
+Cache definition::
+Do not use to mean "cache configuration".
+
+Cache Manager::
+Interface for creating and managing caches. Always spell as two words with capital letters when referring to the abstract notion of a Cache Manager. +
+When referring to specific interfaces, use `CacheManager`, `EmbeddedCacheManager`, or `RemoteCacheManager`.
+
+Clear::
+Delete all the elements of a container. For example, "clear entries in a cache".
+
+Create::
+Build something new. For example, "create a cache".
+
+Delete::
+Remove a single element or a group of similar elements from a container. For example, "delete an entry from the cache" or "delete all expired entries from the cache".
+
+Destroy::
+Do not use to mean "delete".
+
+Index::
+Create an index to improve query performance or add an entry to an existing index when it is added to a cache.
+
+Indexes::
+Plural of index. Do not use "indices".
+
+Purge::
+Can have negative connotations and is not a preferred term. Use only as a synonym for "clear" or "delete" in subsequent descriptions, such as in the body text of a dialog or text that follows a heading. Do not use as a primary verb or in headings.
+
+Query::
+Look up information within the data set. The words "query" and "search" are synonymous but frequently interchanging the two can lead to inconsistency.
++
+Use in verb form: "Query values in the cache." +
+Use in noun form preceded by Ickle: "Perform an Ickle query on the cache."
+
+Reindex::
+Rebuild an index to update or change an existing index. Do not hyphenate (always "reindex" never "re-index").
+
+Remove::
+Use as a synonym for "clear" or "delete" in subsequent descriptions, such as in the body text of a dialog or text that follows a heading. Do not use as a primary verb or in headings.
+
+Reset::
+Restore an initial value or state. For example, "Restore a counter."
+
+Search::
+Synonymous with "query" but not preferred. In verb form, use with common phrases like "full-text search" in conjunction with {brandname} Query API. In noun form when referring to the "org.infinispan.query.Search" entry point.
++
+Use in verb form: "Perform full-text searches through the {brandname} Query API."
+Use in noun form: "Use the {brandname} `Search` entry point to construct Ickle queries."
+
+== {brandname} methods and corresponding verbs
+
+When describing an action that the user performs that invokes an underlying method, use verbs as follows:
+
+|===
```

Comment:
is this table an example? or is it supposed to be a full list and include methods like `cache.remove()`?

Reply:
I've been debating if it would be worth the time to go through and do a full list. I started doing that but it seemed a little redundant with the preceding terminology list. I added `cache.remove()`. What do you think? Worth going through the full list?

Reply:
nah, let's fill it lazily eh eh
when you find a wrong verb being used for a specific method, add it to the list :)